toolkit/tools/internal/sliceutils/sliceutils.go

```diff
@@ -119,3 +119,14 @@ func ContainsFunc[K any](inputSlice []K, fn func(K) bool) bool {
 	}
 	return false
 }
+
+func FindValueFunc[K any](inputSlice []K, fn func(K) bool) (K, bool) {
```

Comment:
Huh, surprised this doesn't have an implementation in the new slice library... Will be nice to swap some of these out.

Not sure what to put here, this feels a bit wordy but...
```suggestion
FindValueFunc returns the first value in the slice that matches fn(item) == true along with true, otherwise it returns the default value for an instance of value's type, and false.
func FindValueFunc[K any](inputSlice []K, fn func(K) bool) (K, bool) {
```